### ISSUE TITLE
Potential Path Traversal

### ISSUE THREAD

**Issue by edainwestoc** [akka/akka#18645]
settings.QueuePath is used to create a new queue without checking the contents of QueuePath:
FileBasedMailbox.scala, lines 34-39:
34  private val queue = try {
35    (new java.io.File(settings.QueuePath)) match {
36      case dir if dir.exists && !dir.isDirectory => throw new IllegalStateException("Path already occupied by non-directory " + dir)
37      case dir if !dir.exists                    => if (!dir.mkdirs() && !dir.isDirectory) throw new IllegalStateException("Creation of directory failed " + dir)
38      case _                                     => // All good
39    }

**Comment by ktoso** [comment]
Hi @edainwestoc,
`FileBasedMailbox` was deprecated during `2.3.x` and is gone from `2.4.x`, we're not planning to invest time into them – instead use Akka Persistence, the very much supreme successor of durable mailboxes :)

**Comment by edainwestoc** [comment]
Konrad,First...let me introduce myself. I am a Sr Application Security Engineer (sorry...;8^)) responsible for the assessment of the code being written for a security standard for the connected car industry. The developers of the code for the proof of concept (at the PoC stage at this point) have used Akka in their development and, as such, I am obliged to try and evaluate its source code.
After I received your response I reached out to the project manager and asked which version of Akka they were using. I just received the response and its version 2.3.14. I will follow up and find out whether they are using any of the deprecated file based methods of which I wrote up one example.
At this point I have a spreadsheet of findings on the Akka library and I was wondering whether entering it via GitHub would be the best approach to addressing the issues. I don't want to raise any unnecessary concerns (depending on your responses) by posting the findings on a public forum like the GitHub.
Thanks,Ed AguinagaSr Application Security EngineerRaytheon/Foreground SecurityeTrans 2020


Date: Mon, 5 Oct 2015 12:56:20 -0700
